src/community/wmts-multi-dimensional/src/main/java/org/geoserver/gwc/wmts/dimensions/DimensionsUtils.java

```diff
@@ -0,0 +1,159 @@
+/* (c) 2016 Open Source Geospatial Foundation - all rights reserved
+ * This code is licensed under the GPL 2.0 license, available at the root
+ * application directory.
+ */
+package org.geoserver.gwc.wmts.dimensions;
+
+import org.geoserver.catalog.*;
+import org.geoserver.gwc.wmts.Tuple;
+import org.geoserver.util.ISO8601Formatter;
+import org.geoserver.wms.WMS;
+import org.geotools.util.Converters;
+import org.geotools.util.Range;
+
+import java.io.Serializable;
+import java.util.*;
+
+/**
+ * Some utils methods useful to interact with dimensions.
+ */
+public final class DimensionsUtils {
+
+    /**
+     * Helper method that will extract a layer dimensions.
+     */
+    public static List<Dimension> extractDimensions(WMS wms, LayerInfo layerInfo) {
+        ResourceInfo resourceInfo = layerInfo.getResource();
+        if (resourceInfo instanceof FeatureTypeInfo) {
+            return extractDimensions(wms, layerInfo, (FeatureTypeInfo) resourceInfo);
+        }
+        if (resourceInfo instanceof CoverageInfo) {
+            return extractDimensions(wms, layerInfo, (CoverageInfo) resourceInfo);
+        }
+        return Collections.emptyList();
+    }
+
+    /**
+     * Helper method that will extract the dimensions from a feature type info.
+     */
+    private static List<Dimension> extractDimensions(WMS wms, LayerInfo layerInfo, FeatureTypeInfo typeInfo) {
+        List<Dimension> dimensions = new ArrayList<>();
+        DimensionInfo timeDimension = typeInfo.getMetadata().get(ResourceInfo.TIME, DimensionInfo.class);
+        if (timeDimension != null) {
+            dimensions.add(new VectorTimeDimension(wms, layerInfo, timeDimension));
+        }
+        DimensionInfo elevationDimension = typeInfo.getMetadata().get(ResourceInfo.ELEVATION, DimensionInfo.class);
+        if (elevationDimension != null) {
+            dimensions.add(new VectorElevationDimension(wms, layerInfo, elevationDimension));
+        }
+        return dimensions;
+    }
+
+    /**
+     * Helper method that will extract the dimensions from a coverage type info.
+     */
+    private static List<Dimension> extractDimensions(WMS wms, LayerInfo layerInfo, CoverageInfo typeInfo) {
+        List<Dimension> dimensions = new ArrayList<>();
+        for (Map.Entry<String, Serializable> entry : typeInfo.getMetadata().entrySet()) {
+            String key = entry.getKey();
+            Serializable value = entry.getValue();
+            if (key.equals(ResourceInfo.TIME)) {
+                dimensions.add(new RasterTimeDimension(wms, layerInfo, Converters.convert(value, DimensionInfo.class)));
```

Comment:
Correct ... for vector types I delegate the dimensions domain values retrieving on the WMS already existing methods (which correctly check if the dimension is enable or not) ... but for rasters I could not found methods for doing that in the WMS (in a cleaning way) and I forgot this use case ... thanks for the patch.